examples/noline.rs

```diff
@@ -0,0 +1,248 @@
+extern crate menu;
+
+use embedded_io::{ErrorType, Read as EmbRead, Write as EmbWrite};
+use menu::*;
+use noline::builder::EditorBuilder;
+use std::{
+    io::{self, Read as _, Stdin, Stdout, Write as _},
+    iter::repeat,
+};
+use termion::raw::IntoRawMode;
+
+pub struct IOWrapper {
+    stdin: Stdin,
+    stdout: Stdout,
+}
+
+impl IOWrapper {
+    pub fn new() -> Self {
+        Self {
+            stdin: std::io::stdin(),
+            stdout: std::io::stdout(),
+        }
+    }
+}
+
+impl Default for IOWrapper {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl ErrorType for IOWrapper {
+    type Error = embedded_io::ErrorKind;
+}
+
+impl EmbRead for IOWrapper {
+    fn read(&mut self, buf: &mut [u8]) -> Result<usize, Self::Error> {
+        self.stdin
+            .read_exact(&mut buf[0..1])
```

Comment:
Why is this limited to one byte? It'll also panic if the caller passes a zero length slice.

Reply:
For no good reason. I'll fix it.